### DISCUSSION TITLE
Capability for CC refresh and Configuration Parameter Change via message to MQTT

### DISCUSSION THREAD

**Discussion by apella12** [zwave-js/zwave-js-ui#3172] [Q&A] [answered]
I have setup zwave-js-ui as a Zwave provider to OpenHAB. I was intrigued by the S2 security, OTA updates, possible certification, and it seems you have a close relationship with Zooz. I don't have the skills to integrate zwave-js so it is kind of a bodge as I need to interface with two programs. My aim was to keep my rules and other bindings on OpenHAB as I have used that program for some time and have a lot invested. I have all (OpenHAB, MQTT and Zwave-js-ui) running on a 2GB RPi4 using a Zooz 700 controller and 34 devices. At this point I have linked the "zwave" section messages of Zwave-js-ui to the appropriate OpenHAB items.  I tried the homeassistant side, but some of the configurations are not supported by jinjava. I can switch (pretty much) between the bindings by stopping/starting the Zwave binding in OpenHAB and enabling/disabling Zwave-js-ui since I'm using the same Z-stick. There are a few quirks I need to investigate

As to my questions; in OpenHAB I have a rule to Refresh the meter CC (50) right before midnight to get a daily KWH.  Is it possible to publish a message to MQTT that will cause that to happen or can it only be refreshed from Zwave-js-ui?  Also in OpenHAB I change the reporting frequency for energy using Configuration CC if the device is running. Again can that be done or triggered via MQTT or just via the UI?

**Comment by AlCalzone** [comment]
MQTT is handled by Z-Wave JS UI, transferring this discussion.
@robertsLando ?

**Comment by robertsLando** [comment]
That's quite easy by using MQTT Apis: https://zwave-js.github.io/zwave-js-ui/#/guide/mqtt?id=refreshccvalues

**Comment by apella12** [comment]
Home automation is part of my retirement plan to keep learning and after OpenHAB for several years this is pretty new to me.  I can't quite follow the documentation.  I think the topic (for the cc refresh) should be;
`zwave/_CLIENTS/ZWAVE_GATEWAY-zwave-js-ui/api/sendCommand/set //Assuming sendCommand is used for all API methods?`
but I'm having trouble seeing how to integrate the refreshccvalues with the example of sendCommand. Is the payload  `{"args":[{"nodeId": 24,"commandClass":50, "endpoint":0},"refreshCCValues"]}` (this was a valid JSON) or do I need to include the command args `{"args":[{"nodeId": 24,"commandClass":50, "endpoint":0},"refreshCCValues",[{(nodeId: 24, cc: 50): Promise<void>;}]]}` (this does not seem to be a valid JSON).  I'm probably going to need an example.

On the plus side I have the parameter change working, so that is resolved.

One other item however, again related to zwave traffic.  Can the "GET" 5 seconds after the SET be changed on a node by node basis?  I have that suppressed with OpenHab.

EDIT: I go this to pass the JSON validator, but not sure if this is what is needed.
```
{
	"args": [{
		"nodeId": 24,
		"commandClass": 50,
		"endpoint": 0  //IS this needed for Command Class refresh??
	}, "refreshCCValues", [{
		"nodeId": 24,
		"cc": 50
	}, "Promise<void>"]]
}
```

**Reply by robertsLando** [reply, marked as answer]
@apella12 I already sent you the link that pointed to the `refreshCCValues` api. That's easier to do, sendCommand api is more generic and allows you to send all supported CC commands but when there is a dedicated api I suggest you to use that instead.

Check https://zwave-js.github.io/zwave-js-ui/#/guide/mqtt?id=refreshccvalues

Topic
```
zwave/_CLIENTS/ZWAVE_GATEWAY-zwave-js-ui/api/refreshCCValues/set
```

Payload
```
{
    "args": [
        24,
        50
    ]
}
```

**Reply by apella12** [reply]
Thanks!.  FYI-As a NOOB I never would have gleaned that answer from the example.  I did go to the link, but thought the API needed to be added to the sendCommand.  Also I was thrown because the API documentation was not in JSON and was unfamiliar with the promise<void> parameter. Anyway I have this working in the MQTT explorer, just need to figure out how to get the command out of OpenHAB
